addons/event/models/event_event.py

```diff
@@ -402,7 +402,9 @@ def _compute_event_mail_ids(self):
                 continue
 
             # lines to keep: those with already sent emails or registrations
-            mails_toremove = event._origin.event_mail_ids.filtered(lambda mail: not mail.mail_sent and not(mail.mail_registration_ids))
+            mails_toremove = event.event_mail_ids.filtered(
```

Comment:
Could be great to have a test case showing what is fixed, because I am quite confused with the change.